x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java

```diff
@@ -447,7 +430,7 @@ private void retrieveStackTraces(
         GetStackTracesResponseBuilder responseBuilder,
         ActionListener<GetStackTracesResponse> submitListener
     ) {
-        if (mayNotifyOfCancellation(submitTask, submitListener)) {
+        if (submitTask instanceof CancellableTask c && c.notifyIfCancelled(submitListener)) {
```

Comment:
I think `submitTask` is _always_ a `CancellableTask`, it'd be best to change the argument type to reflect this (and cast it to `CancellableTask` in `doExecute` instead).

Reply:
Thanks! I've pushed https://github.com/elastic/elasticsearch/pull/107139.